packages/replayio/src/utils/prompt/prompt.ts

```diff
@@ -0,0 +1,38 @@
+import { readFromCache, writeToCache } from "../cache";
+import { promptHistoryPath } from "./config";
+import { PromptHistory } from "./types";
+
+export async function prompt(id?: string): Promise<boolean> {
+  return new Promise(resolve => {
+    const stdin = process.stdin;
+    const wasRaw = stdin.isRaw;
+    stdin.setRawMode(true);
+    stdin.resume();
+    stdin.setEncoding("utf8");
+
+    function onData(data: string) {
+      stdin.off("data", onData);
+      stdin.setRawMode(wasRaw);
+      stdin.setEncoding();
+
+      if (id) {
+        const cache = readFromCache<PromptHistory>(promptHistoryPath) ?? {};
+        writeToCache<PromptHistory>(promptHistoryPath, {
+          ...cache,
+          [id]: Date.now(),
+        });
+      }
+
+      switch (data) {
+        case "\r":
+          resolve(true);
+          break;
+        default:
+          resolve(false);
+          break;
+      }
+    }
+
+    stdin.on("data", onData);
```

Comment:
TODO: it would be great to recheck if exit signals etc are still handled correctly with this. I'm not familiar with the in and outs of those node APIs so this very well might be already working alright - just a thought about what might be worth rechecking here.